Applications/lib/python/mirtk/batch/condor.py

```diff
@@ -45,7 +45,11 @@ def submit(name, command=None, args=[], opts={}, script=None, tasks=0, deps=[],
                 log = os.path.join(logdir, name + "_$(Cluster).$(Process).log")
             else:
                 log = os.path.join(logdir, name + "_$(Cluster).log")
-        os.makedirs(logdir)
+        try:
+            os.makedirs(logdir)
+        except OSError:
+            # no error for dir already exists
+            pass
```

Comment:
Thanks, @jcupitt for this fix!

Sorry I'm looking at this so much later... I've seen in `slurm.py`, I've added the `os.path.exists` check. You are right, that exception handling avoids the risk of race conditions and is probably better. The only problem with the suggested change is that you ignore all `OSError` exceptions, not only the one due to already existing directory.

Better:
```
try:
    os.makedirs(logdir)
except OSError as exc:
    if exc.errno != errno.EEXIST:
        raise
    pass
```

See also: https://stackoverflow.com/questions/18973418/os-mkdirpath-returns-oserror-when-directory-does-not-exist

Given that it's quite some time now, do you still want to update the PR? I'm happy to merge it then. Otherwise, I can make the changes myself as well. If you do update the PR, would you also consider changing the 

```
if not os.path.exists(logdir):
    os.makedirs(logdir)
```
lines in `slurm.py` to do it the same way?

Reply:
I've just discovered there is a `mirtk.utils.mkdirs` helper that encapsulates this logic. Thus, all `os.makedirs` calls should be replaced by `from mirtk.utils import mkdirs` rather.

https://github.com/BioMedIA/MIRTK/blob/ec290287dadb2bf11134f22ba6e0ae889c52de43/Applications/lib/python/mirtk/utils.py#L27

Reply:
Good grief. OK, that's useful, I'll check for all `makedirs()` calls.

Reply:
Heh, I found another one.

https://github.com/MIRTK/Deformable/blob/master/lib/python/mirtk/deformable/neonatal_cortex.py#L130

Is it OK for stuff in `Packages` to depend on stuff in `Applications`? If so, I'll swap that for the `Applications` helper too.

Reply:
Yeah, good question. I think it's best to keep the package-local definition of `mkdirs` of both Applications and Deformable package. There's a Scripting package that could provide such utilities, but there's not much in it yet. So for now, I think it's good as is after your change today.